### PR TITLE
smb: add smb.version keyword 5075 v9

### PR DESCRIPTION
**Pull request by jmtaylor90**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5075

Describe changes:
- removed unnecessary smb::SMBTransaction include
- updated documentation example rule formatting
- updated keyword description to include where version is determined
- fixed dod pointer leak
-

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=pr/1380
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Review by jufajardini — CHANGES_REQUESTED** [review]
I see that stuff from previous reviews seems to have been incorporated, thanks!

Sorry for catching some new stuff in this one >__<'

File: doc/userguide/rules/smb-keywords.rst [hunk 1]
```diff
@@ -61,6 +61,56 @@ Examples::
...
+
+- 0xffSMB is SMB1 `header <https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-smb/3c0848a6-efe9-47c2-b57a-f7e8217150b9>`_
+- 0xfeSMB is SMB2 `normal header <https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-smb2/5cd64522-60b3-4f3e-a157-fe66f1228052>`_ (can be `sync <https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-smb2/fb188936-5050-48d3-b350-dc43059638a4>`_ or `async <https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-smb2/ea4560b7-90da-4803-82b5-344754b92a79>`_)
+- 0xfdSMB is SMB2 `transform header <https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-smb2/d6ce2327-a4c9-4793-be66-7b5bad2175fa>`_. This is only valid for the SMB 3.x dialect family.
+- 0xfcSMB is SMB2 `transform compression header <https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-smb2/d6ce2327-a4c9-4793-be66-7b5bad2175fa>`_ (can be `chained <https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-smb2/aa880fe8-ebed-4409-a474-ec6e0ca0dbcb>`_ or `unchained <https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-smb2/793db6bb-25b4-4469-be49-a8d7045ba3a6>`_). These ones requires the use of 3.1.1 dialect.
```
[the remark this answers is not in the document]
Comment (jmtaylor90) [reply doc/userguide/rules/smb-keywords.rst:98]: Updated in the next PR

File: doc/userguide/rules/smb-keywords.rst [hunk 2]
```diff
@@ -61,6 +61,56 @@ Examples::
...
+
+The Protocol ID in the header distinguishes only SMB1 and SMB2 since they are totally different protocols with total different message formats, types and implementation.
```
[the remark this answers is not in the document]
Comment (jmtaylor90) [reply doc/userguide/rules/smb-keywords.rst:100]: Updated in the next PR

File: doc/userguide/rules/smb-keywords.rst [hunk 3]
```diff
@@ -61,6 +61,56 @@ Examples::
...
+- 2.0.2
+- 2.1
+- 3.0
+- 3.0.2
+- 3.1.1
+
+We say we are using SMB3 when we select a 3.x dialect for the conversation, so you can use SMB3.0, SMB3.0.2 or SMB3.1.1. The higher you choose, the more capabilities you have, but the message syntax and message command number remains the same.
+
+SMB version and dialect are separate components. In the case of SMBv3 for instance, the SMB version will be 2 but the dialect will be 3.x.
+
```
[the remark this answers is not in the document]
Comment (jmtaylor90) [reply doc/userguide/rules/smb-keywords.rst:113]: Updated to include that information in the next PR

File: src/detect-engine-register.c [hunk 4]
```diff
@@ -611,6 +612,8 @@ void SigTableSetup(void)
     DetectSmbShareRegister();
     DetectSmbNtlmsspUserRegister();
     DetectSmbNtlmsspDomainRegister();
+    DetectSmbVersionRegister();
+
```
[the remark this answers is not in the document]
Comment (jmtaylor90) [reply src/detect-engine-register.c:616]: Updated in next PR

**Pull request closed without merging**